virtualizarr/readers/dmrpp.py

```diff
@@ -0,0 +1,683 @@
+        raise ValueError("Not enough information to parse Dim/Dimension tag")
+
+    def _parse_multi_dims(
+        self, dim_tags: list[ET.Element], global_dims: dict[str, int] = {}
+    ) -> dict:
+        """
+        Parse multiple <Dim> or <Dimension> tags. Generally <Dim> tags are found within dmrpp variable tags.
+
+        Returns best possible matching of {dimension: shape} present in the list and global_dims. E.g tags=(Dim("lat", None), Dim("lon", None)) and global_dims={"lat": 100, "lon": 100, "time": 5} --> {"lat": 100, "lon": 100}
+
+        E.g. tags=(Dim("time", None), Dim("", 200)) and global_dims={"lat": 100, "lon": 100, "time": 5} --> {"time": 5, "phony_dim0": 200}
+
+        This function is often used to fill in missing sizes from the global_dims. E.g. Variable tags may contain only dimension names and not sizes. If the {name: size} matching is known from the global_dims, it is used to fill in the missing sizes.
+
+        Parameters
+        ----------
+        dim_tags : tuple[ET.Element]
+            A tuple of ElementTree Elements representing dimensions in the DMR file.
+
+        global_dims : dict
+            A dictionary of dimension names and sizes. E.g. {"time": 1, "lat": 1447, "lon": 2895}
+
+        Returns
+        -------
+        dict
+            E.g. {"time": 1, "lat": 1447, "lon": 2895}
+        """
+        dims: dict[str, int | None] = {}
+        for dim_tag in dim_tags:
+            dim: dict[str, int | None] = self._parse_dim(dim_tag)
+            if "phony_dim" in dim:
+                dims["phony_dim_" + str(len(dims))] = dim["phony_dim"]
+            else:
+                dims.update(dim)
+        for name, size in list(dims.items()):
+            if name in global_dims and size is None:
+                dims[name] = global_dims[name]
+        return dims
+
+    def _parse_variable(
+        self, var_tag: ET.Element, dataset_dims: dict[str, int]
+    ) -> xr.Variable:
+        """
+        Parse a variable from a DMR tag.
+
+        Parameters
+        ----------
+        var_tag : ET.Element
+            An ElementTree Element representing a variable in the DMR file. Will have DAP dtype as tag.
+
+        dataset_dims : dict
+            A dictionary of dimension names and sizes. E.g. {"time": 1, "lat": 1447, "lon": 2895}
+            Must contain at least all the dimensions used by the variable. Necessary since the variable
+            metadata only contains the dimension names and not the sizes.
+
+        Returns
+        -------
+        xr.Variable
+        """
+        # Dimension names
+        dim_tags = var_tag.findall("dap:Dim", self._ns)
+        dim_shapes = self._parse_multi_dims(dim_tags, dataset_dims)
+        # convert DAP dtype to numpy dtype
+        dtype = np.dtype(
+            self._dap_np_dtype[var_tag.tag.removeprefix("{" + self._ns["dap"] + "}")]
+        )
+        # Chunks and Filters
+        filters = None
+        shape: tuple[int] = tuple(dim_shapes.values())
```

Comment:
See https://github.com/zarr-developers/VirtualiZarr/actions/runs/10561663864/job/29257886654

```suggestion
        shape: tuple[int, ...] = tuple(dim_shapes.values())
```